src/wayland/compositor/mod.rs

```diff
@@ -453,6 +458,21 @@ pub struct CompositorClientState {
     queue: Mutex<Option<TransactionQueue>>,
 }
 
+impl CompositorClientState {
+    pub fn blocker_cleared<D: CompositorHandler>(
+        &self,
+        surface: &WlSurface,
+        state: &mut D,
+        dh: &DisplayHandle,
+    ) {
+        if let Some(queue) = self.queue.lock().unwrap().as_mut() {
```

Comment:
Is it possible that commit again tries to access the queue? That could lead to a deadlock, which could be prevented if the lock is dropped before invoking commit.

Reply:
:+1: There is no reason to hold the lock while `commit()` is invoked, and I suggest we stick to a "hold lock as shortly as needed" policy.